File: Python Fundamentals/4. Functions/perfect_number.py
def check_is_perfect_number(num):
    half = num // 2
    nums = [el for el in range(half, 0, -1) if num % el == 0]
    if num == sum(nums):
        return "We have a perfect number!"
    return "It's not so perfect."


num = int(input())
print(check_is_perfect_number(num))

File: JavaScript Fundamentals/JavaScript Fundamentals Functions/perfectNumber.js
function isPerfectNumber(num) {
    let sum = 0;
    for (let i = 1; i <= num / 2; i++) {
        if (num % i === 0) {
            sum += i;
        }
    }

    if (sum === num && sum !== 0) {
        console.log('We have a perfect number!');
    } else {
        console.log("It's not so perfect.");
    }
}

isPerfectNumber(6);
isPerfectNumber(28);
isPerfectNumber(1236498);